src/settings.cpp

```diff
@@ -859,6 +811,82 @@ QString directoryAddress = GetIniSetting ( IniXMLDocument, "server", "centralser
         pServer->SetRecordingDir ( FromBase64ToString ( GetIniSetting ( IniXMLDocument, "server", "recordingdir_base64" ) ) );
     }
 
+    // to avoid multiple registrations, must do this after collecting serverinfo
+    if ( !CommandLineOptions.contains ( "--centralserver" ) && !CommandLineOptions.contains ( "--directoryserver" ) )
+    {
+        // custom directory
+        // CServerListManager defaults to command line argument (or "" if not passed)
+        // Server GUI defaults to ""
+        QString directoryAddress = "";
+        // clang-format off
+// TODO compatibility to old version < 3.8.2
+directoryAddress = GetIniSetting ( IniXMLDocument, "server", "centralservaddr", directoryAddress );
+        // clang-format on
```

Comment:
Not a huge fan of this break-identation-to-get-attention scheme personally (I know that the original code uses this in several places).

Reply:
> in several places

"Consistently" might be a better way of putting it ;).